### DISCUSSION TITLE
Python Scripts with SofaUnity

### DISCUSSION THREAD

**Discussion by HaydenGowing** [InfinyTech3D/SofaUnity#80] [Help - Using SofaUnity]
Hello, 

I am working on a project to model and display breast with a tumor being deformed by an ultrasound probe through SOFA and using Unity to display it through a HoloLens VR headset. All of our code is in python and we are using the SofaPython plugin as we have to import quite a lot of libraries and files in order to develop this simulation. I have downloaded SofaUnity and opened the examples on Unity but I can't find any way to import a python file. Is important a python file possible or does it have to be an XML code file (.scn)? 

Thank you

**Comment by epernod** [comment]
Hi,

the Sofa-Python support has been broken. We are working on it to restore this feature. It should be available in next release (v24.06) coming soon.
Are you using python script controller from your SOFA scene?

Best regards

**Reply by dream520nb** [reply]
Hello, I am also having problems importing xml files
![image](https://github.com/user-attachments/assets/42c0ed13-f58f-4582-9149-244084f881b7)
![image](https://github.com/user-attachments/assets/59872a8b-d432-4bc7-b4fa-bd5b581af812)
![image](https://github.com/user-attachments/assets/df58d8ac-05d1-40de-bf0d-52a23e0ff700)
![image](https://github.com/user-attachments/assets/2a7c8e10-98da-4cfc-aad5-8cdfd48d6425)

**Comment by epernod** [comment]
Hi,

The new release v24.06 has been released: https://github.com/InfinyTech3D/SofaUnity/releases/tag/SofaUnity_v24.06.00
Could you test again with this new version.

Thank you
Best regards

**Reply by epernod** [reply]
any update @HaydenGowing 
thank you
